### PR TITLE
Remove duplicate discrete keys

### PR DESCRIPTION
**Pull request by varunagrawal**
This PR removes duplicate keys from `DCFactorGraph::discreteKeys()` as well as from the printing of DCMixtureFactor.

I want to add an extra unit test for this which I will do in a bit.

### CODE REVIEW

**Comment by varunagrawal** [comment]
Ping!

**Comment by dellaert** [comment]
ping @ProfFan

**Comment by varunagrawal** [comment]
Ping @ProfFan!

File: gtsam/hybrid/DCFactorGraph.h [hunk 1]
```diff
@@ -39,6 +39,10 @@ class DCFactorGraph : public FactorGraph<DCFactor> {
       }
     }
 
+    // STL trick to remove duplicate keys
+    std::sort(result.begin(), result.end());
```
Comment (ProfFan) [inline gtsam/hybrid/DCFactorGraph.h]: Is it the correct thing to do? I think the order of the keys indicate the ordering used to eliminate, is it?

**Review by ProfFan — APPROVED** [review]
Approving sans one comment.

File: gtsam/hybrid/DCFactorGraph.h (hunk 1 above)
Comment (varunagrawal) [reply gtsam/hybrid/DCFactorGraph.h]: Good point. I wasn't too happy about sorting either so let me see if I can use something else.

Comment (varunagrawal) [reply gtsam/hybrid/DCFactorGraph.h]: Updated!

**Pull request merged**